.github/workflows/linux.yml

```diff
@@ -34,23 +40,54 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y libopenblas-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache libmpc-dev
 
+    - name: Caching Python dependencies
+      uses: actions/cache@v2
+      id: cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
     - name: Install Python packages
       run: |
         python -m pip install numpy setuptools wheel cython pytest pytest-xdist pybind11 pytest-xdist mpmath gmpy2 pythran ninja
```

Comment:
@rgommers I had a question on this part. Original comment here: https://github.com/HarshCasper/scipy-fork/commit/ae7cc2be6f4bdc28d23643d4e9194da051d2ee27#r56246707

Should we try to install the Python packages via a `requirements.txt` or `pyproject.toml` ?

Reply:
answered on the original comment, so resolving here